File: experiments/016-LanguageComparisonRepeated/cells/java-gradle_1-Empty_free/2022-1c-recuperatorio-parcial1/20260917-060211-45590/output/project/src/main/java/customerimporter/NewZipCode.java
package customerimporter;

import java.util.function.Supplier;

public class NewZipCode extends ZipCode {

    // instance creation

    public static boolean canHandle(String aZipCode) {
        return Character.isLetter(aZipCode.charAt(0));
    }

    // error messages

    public static String invalidNewZipCodeErrorDescription() {
        return "Invalid new zipcode";
    }

    // initialization

    public NewZipCode(String aZipCode) {
        super(aZipCode);
    }

    // accessing

    public Object value() {
        return code;
    }

    // testing

    public boolean isOld() {
        return false;
    }

    public boolean isNew() {
        return true;
    }

    public Object oldZipCodeIfNone(Supplier<Object> aNoneBlock) {
        return aNoneBlock.get();
    }

    public Object newZipCodeIfNone(Supplier<Object> aNoneBlock) {
        return value();
    }

    // validation

    // "B1636BBE": a letter, the old zip code and three letters
    protected void assertIsValid() {
        assertThat(code.length() == 8);
        String oldZipCode = code.substring(1, 5);
        assertThat(Characters.areAllDigits(oldZipCode));
        assertThat(Integer.parseInt(oldZipCode) > 999);
        assertThat(Characters.areAllLetters(code.substring(code.length() - 3)));
    }

    protected String invalidZipCodeErrorDescription() {
        return invalidNewZipCodeErrorDescription();
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2022-1c-recuperatorio-parcial1/20260921-202259-53403/workdir/src/customerimporter/new_zip_code.py
from customerimporter.zip_code import ZipCode


class NewZipCode(ZipCode):

    # type

    @classmethod
    def can_handle(cls, a_zip_code):
        return a_zip_code[0].isalpha()

    # error messages

    @classmethod
    def invalid_value_error_description(cls):
        return "Invalid new zipcode"

    # validation

    @classmethod
    def assert_is_valid(cls, a_zip_code):
        # "B1636BBE" size 8
        cls.assert_size_is(a_zip_code, 8)
        cls.assert_is_valid_zip_code_number(a_zip_code[1:5])
        cls.assert_all_are_letters(a_zip_code[-3:])

    # testing

    def is_new(self):
        return True

    # zip code

    def new_zip_code_if_none(self, a_none_block):
        return self.value()
